### PR TITLE
Alien infestation larva spawning (sane edition)

### PR DESCRIPTION
**Pull request by Tyranicranger4**
## About The Pull Request

Changes larva in the alien infestation event to spawn on xeno_spawn landmarks instead of on a random unwelded vent.

Generally improves the placement of xeno_spawn markers ensuring that they have access to a vent/scrubber.
As part of the above this PR also adds extra vents to maintenance in a few locations. Including but not limited to scrubbers/vents in solar maintenance on both Fland and Box.

## Why It's Good For The Game

Makes it significantly less likely for larva to spawn in view of someone.
As it is right now there's nothing stopping a larva from dropping in the main hallway, chemistry, or other very visible locations.

Better placed xeno_spawn markers should mean you always spawn with near immediate access to a vent/scrubber.
## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>
Triggered the event, spawned where you'd expect.

image
image
image
image
image
image

</details>

## Changelog
:cl:
tweak: Larva from the 'alien infestation' event now spawn on xeno_spawn landmarks instead of random unwelded vents.
tweak: Improved xeno_spawn landmark placement on all maps.
add: Added a few extra vents in maintenance rooms on all maps.
add: Added scrubbers/vents to solars maintenance on Fland and Box.

/:cl:

### CODE REVIEW

**Comment by Tyranicranger4** [comment]
Worth noting that not all of the xeno_spawn landmarks have immediate access to a vent. If that's seen as a major issue I have no problems relocating the spawns.

**Comment by PowerfulBacon** [comment]
A lot of the xeno spawns have awful placements, if you spawn in maints then you often aren't next to a vent and will be screwed

**Comment by Tyranicranger4** [comment]
Yeah, kinda worried about that. I'll replace the spawns.

**Comment by PowerfulBacon** [comment]
I think I placed most of the ones on corg station near vents, so that should be not horrible at the very least. Not sure about other maps, they might have done the same

**Comment by EvilDragonfiend** [comment]
now put a wall on those and larva will never escape

**Comment by Tyranicranger4** [comment]
> now put walls around on those and larva will never escape

Currently you could build 400 vents in an isolated box connected to at least 20 pipes and larva will never escape :trollface:

**Comment by EvilDragonfiend** [comment]
> > now put walls around on those and larva will never escape
> 
> Currently you could build 400 vents in an isolated box connected to at least 20 pipes and larva will never escape :trollface:

Based :trollface:

**Review by itsmeow — CHANGES_REQUESTED** [review]
You need to update the dynamic version as well

File: code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm [hunk 1]
```diff
@@ -409,25 +409,14 @@
...
-			continue
-		if(is_station_level(temp_vent.loc.z) && !temp_vent.welded)
-			var/datum/pipeline/temp_vent_parent = temp_vent.parents[1]
-			if(!temp_vent_parent)
-				continue // No parent vent
-			// Stops Aliens getting stuck in small networks.
-			// See: Security, Virology
-			if(length(temp_vent_parent.other_atmosmch) > 20)
-				vents += temp_vent
-	if(!length(vents))
```
Comment (itsmeow) [inline code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm:423]: Delete `var/list/vents` from the rule since it's now unused

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm (hunk 1 above)
Comment (Tyranicranger4) [reply code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm:423]: Certified blind moment.

**Comment by Tyranicranger4** [comment]
Mapping conflict jumpscare.

**Review by itsmeow — DISMISSED** [review]

**Comment by Tyranicranger4** [comment]
Oh fuck you merged the decal thing. I'll fix this when I'm less sleep deprived then.

**Comment by Tyranicranger4** [comment]
Don't care enough to fix conflicts and whatever happened to the decals on Kilo :trollface:

**Pull request closed without merging**